### ISSUE TITLE
a.b['c'] plugin matching

### ISSUE THREAD

**Issue by mishk0** [webpack/webpack#1997]
Is it possible to catch such call `a.b['c']()` in a plugin? This way `call a.b.c`([docs](https://webpack.github.io/docs/plugins.html#call-identifier-bailing)) doesn't work

**Comment by IngwiePhoenix** [comment]
To start, `a.b.['c']` is invalid syntax.

- get property `b` from `a` (`a.b`) is valid. But...
- What does `.['c']` mean? `[]` is an index-accessor operator - which you can pass an index as string or number. But the period before the brackets is _also_ an index-accessor operator, but for named properties of an object.

```javascript
var a = { foo: "bar" }
a.foo // OK
a["foo"] // OK

var b = [1, 2, 3];
b.0 // Not OK!
b[0] // OK.

b.foo = "bar";
b["foo"] // OK
b.foo // OK

var c = { 0: 1 }
c.0 // OK
c["0"] // OK
```

When attaching a "`call a.b.c`" plugin, you are looking for calls like: `a.b.c()`. I.e.:

```javascript
a.b.c("foo"); // -> Results in callee to be a.b.c, and arguments to be 1, with 1st being a bare string.
```

**Comment by mishk0** [comment]
Thanks, it's a typo, I mean `a.b['c']()`

**Comment by IngwiePhoenix** [comment]
`call a.b.c` should do it, then, I suppose.

**Comment by mishk0** [comment]
if I rewrite it and invoke like that `a.b.c()`, it will work. But in some cases I have to use this notation `a.b['c']`

**Comment by sokra** [comment]
yep it's possible. Would you like to try a PR? We already had an issue with `System["import"]()`..

**Comment by mishk0** [comment]
Ok, I'll look

**Comment by mxstbr** [comment]
Fixed in 7af71d4a6943237f9710a754d7c6cb4a45c0f60c, can be closed.
